File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix,suffix=[0]*len(nums),[0] *len(nums)
        n=len(nums)
        prefix[0]=1
        suffix[n-1]=1
        for i in range(1,n,1):
            prefix[i]=prefix[i-1]*nums[i-1]
        for i in range(n-2,-1,-1):
            suffix[i]=suffix[i+1]*nums[i+1]
        ans=[0]*len(nums)
        for i in range(n):
            ans[i]=prefix[i]*suffix[i]
        return ans
		

File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        vector<vector<int>> ans;
        set<vector<int>> temp;
        for (int k=0;k<n;k++){
            int i,j;
            if(k==0) i=1;
            else i=0;
            if(k==n-1) j=n-2;
            else j=n-1;
            while (i<j){
                if(i==k) {i++; continue;}
                if(j==k) {j--; continue;}
                if(nums[i]+nums[j]<-nums[k]){
                    i++;
                }
                else if (nums[i]+nums[j]>-nums[k]){
                    j--;
                }
                else{
                    vector<int> temp1={nums[i],nums[j],nums[k]};
                    sort(temp1.begin(),temp1.end());
                    temp.insert(temp1);
                    i++;
                }
            }
        }
        ans.assign(temp.begin(),temp.end());
        return ans;
    }
};
